File: common/protcol/c#/caller/gate_call_clientcaller.cs
/*this caller file is codegen by juggle for c#*/
using System;
using System.Collections;
using System.IO;

namespace caller
{
    public class gate_call_client : juggle.Icaller 
    {
        public gate_call_client(juggle.Ichannel _ch) : base(_ch)
        {
            module_name = "gate_call_client";
        }

        public void connect_server_sucess()
        {
            ArrayList _argv = new ArrayList();
            call_module_method("connect_server_sucess", _argv);
        }

        public void ack_heartbeats()
        {
            ArrayList _argv = new ArrayList();
            call_module_method("ack_heartbeats", _argv);
        }

        public void call_client(String argv0,String argv1,ArrayList argv2)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            _argv.Add(argv1);
            _argv.Add(argv2);
            call_module_method("call_client", _argv);
        }

    }
}

File: protcol/js/caller/gate_call_clientcaller.js
/*this caller file is codegen by juggle for js*/
function gate_call_client_caller(ch){
    Icaller.call(this, "gate_call_client", ch);

    this.ntf_uuid = function( argv0){
        var _argv = [argv0];
        this.call_module_method.call(this, "ntf_uuid", _argv);
    }

    this.connect_gate_sucess = function(){
        var _argv = [];
        this.call_module_method.call(this, "connect_gate_sucess", _argv);
    }

    this.connect_hub_sucess = function( argv0){
        var _argv = [argv0];
        this.call_module_method.call(this, "connect_hub_sucess", _argv);
    }

    this.ack_heartbeats = function(){
        var _argv = [];
        this.call_module_method.call(this, "ack_heartbeats", _argv);
    }

    this.call_client = function( argv0, argv1, argv2){
        var _argv = [argv0,argv1,argv2];
        this.call_module_method.call(this, "call_client", _argv);
    }

}
(function(){
    var Super = function(){};
    Super.prototype = Icaller.prototype;
    gate_call_client_caller.prototype = new Super();
})();
gate_call_client_caller.prototype.constructor = gate_call_client_caller;

